internal/db/p2p.go

```diff
@@ -292,3 +292,17 @@ func (db *DB) SyncCollectionVersions(ctx context.Context, versionIDs ...string)
 
 	return txn.Commit()
 }
+
+// SyncBranchableCollection requests the latest version of the branchable collection's DAG
+// from the network and synchronizes it locally. This syncs the collection-level history
+// for branchable collections (collections marked with @branchable directive).
+// It doesn't automatically subscribe to the collection for future updates.
+// context.WithTimeout can be used to set a timeout for the operation.
+//
+// WARNING: This function does not respect transactions.
```

Comment:
thought: Don't worry about this now, idea just popped into my head - for this and the similar funcs, I think we can and should respect transactions (implicit and explicit) for the `datastore` related operations.  The syncing of blocks should be the only bit to ignore a transaction.